binder/binder_option.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binder
+
+type BinderOption interface {
```

Comment:
optional suggestion: `BinderOption` could be `type BinderOption func(*binder)` then the options themselves could just be a simple curry: 

```golang
func WithCredentialFilterFunc(f func(Credentials) error) BinderOption {
    return func(b *binder) {
        b.workloads.credentialFilter = f
    }
}
```